### ISSUE TITLE
App loops infinitely if Groovy configuration throws an exception

### ISSUE THREAD

**Issue by Slugger** [grails/grails-core#10835]
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

```
#!/bin/bash

cd $HOME
grails create-app hard-loop
cd hard-loop
echo "my.config.value = 1" > grails-app/conf/application.groovy
echo "my.bad.config.value = 1 / 0" > grails-app/conf/runtime.groovy
./grailsw run-app
```

### Expected Behaviour

The runtime exception in runtime.groovy should be caught and cause the app to die (division by zero in this example).

### Actual Behaviour

The app startup gets caught in a hard loop, pegging the CPU at 100% until the process is forcibly killed.

### Environment Information

- **Operating System**: Ubuntu 16.04 LTS
- **Grails Version:** grails --version
| Grails Version: 3.3.1
| Groovy Version: 2.4.11
| JVM Version: 1.8.0_131

- **JDK Version:** 1.8.0_131
- **Container Version (If Applicable):** n/a

### Example Application

See script above to create sample app

**Comment by jameskleeh** [comment]
It doesn't matter if the code is in application or runtime groovy

**Comment by jameskleeh** [comment]
The root issue lies in Spring Boot here: https://github.com/spring-projects/spring-boot/issues/10362. It has been fixed in Spring Boot 1.5.8, which will be included in the next 3.3.x release. https://github.com/grails/grails-core/commit/7e88f0d52136e3f5917cf22ffba92b9fb7dad9eb
